packages/engine-test-utils/src/__tests__/engine-server/enginev2.spec.ts

```diff
@@ -189,3 +185,65 @@ describe("engine, config/", () => {
     });
   });
 });
+
+describe("engine, init", () => {
+  test("handles broken frontmatter gracefully", async () => {
+    const dir = tmpDir().name;
+    const fname = "broken-note.md";
+    // Create a workspace with a note that has a broken frontmatter
+    await runEngineTestV5(
+      async ({ wsRoot, vaults }) => {
+        await NoteTestUtilsV4.createNote({
```

Comment:
you can write this in `preSetupHook`. that is for doing any setup ahead of vault initialization